### PR TITLE
Ship v1.97.8: updates can no longer lose lines edited directly into CLAUDE.md

### PR DESCRIPTION
**Pull request by davekilleen**
## Linked Issue
- Linear issue: none — hotfix wave from a beta tester's update-safety report (she reconstructed `_compose_claude`'s output, SHA-matched the preview, and stopped one step short of losing a month of direct CLAUDE.md edits). Founder-gated design for the deeper finding included.

## What Changed
- **CLAUDE.md composition safety (the headline fix):** the composer and the between-updates recompose path now measure the live CLAUDE.md against the installed release's expected composition (activation tag, falling back to the brain's `refs/dex/installed` pin) and refuse to write over any line that exists only in the user's editor. The composer raises `CompositionError` listing the exact lines (bounded at 40, 160-char display truncation) plus the rescue ("move these lines into CLAUDE-custom.md — Dex can do this for you"); the update then keeps CLAUDE.md byte-for-byte with the full reason in the plan's `kept_reasons`. `recompose_if_needed` (normal and `force=True`) returns the same refusal without touching the file. An unprovable baseline is never treated as "no edits".
- **Doctor early warning:** new quick check `config.claude_direct_edits` (OK / BROKEN-with-tier-3-never-applied-heal / UNKNOWN) names the at-risk lines during checkups; the existing `config.claude_composition` deep probe's BROKEN branch now distinguishes direct-edit drift so its advice no longer points at a heal that deliberately refuses.
- **Pillars overlay:** the User Profile section now falls back to `System/pillars.yaml` names when `user-profile.yaml` carries no configured pillars (explicit profile value still wins; a present-but-unreadable pillars file fails closed like the profile itself). Fixes "Pillars: Not yet configured" on configured vaults.
- **Founder-gated design (no code):** `docs/superpowers/specs/2026-09-04-customization-reanchoring-design.md` — a non-circular escape from "release-identity-unproved" exclusions, grounded in code with an empirical repro, recommending a verified release anchor and rejecting trust-on-first-use adoption for v1. Seven decisions queued for the founder; nothing built until ruled.
- **Release:** `scripts/release.sh patch` cut v1.97.8 on the branch (tag pushed after merge); changelog entry in the public house style crediting the reporter.

## Test Plan
- Unit/integration tests added or updated: 25 new tests — detector semantics (line-set, markers, duplicates, non-UTF-8), composer refusal listing exact lines and the 40-line bound, the reporter's exact shape (markers deleted, stale custom file, edits outside any block) on both write paths, no-baseline fail-closed and pass cases, `refs/dex/installed` fallback, all doctor probe verdicts, update-keeps-file end-to-end, pillars fallback/precedence/empty/malformed.
- Negative/error-path tests added or updated: unprovable-baseline refusal, force-recompose refusal, malformed `pillars.yaml` composition refusal.
- Commands run locally: targeted suites 368 passed + full-suite run in progress at PR-open time (orchestrator re-verifies before merge); `ruff check` clean on all touched files; all ten quality-gate scripts green including release-tag capacity and distribution; live injection demo of the reporter's scenario produced the refusal naming her exact lines.

## Ralph Wiggum Loop
- [x] I implemented the change.
- [x] I self-reviewed for defects and edge cases.
- [x] I requested specialist review for risky areas (testing/infra/security when relevant).
- [x] I addressed review findings and re-ran checks.

## Quality Gates
- [x] I added/updated tests or documented why no tests are needed.
- [x] I added a regression test for bug fixes, or this PR is not a bug fix.
- [x] I validated failure modes / edge cases.
- [x] I updated docs or confirmed no docs impact.
- [x] CI checks for lint + tests + coverage are expected to pass.

## Risk & Rollback
- Risk level (low/medium/high): medium — the guard makes two long-standing write paths refuse where they previously overwrote; a false refusal is loud and harmless (the file is kept and the reason listed), while the failure mode it replaces was silent data loss. The pillars change is additive fallback only.
- Rollback plan: revert the merge; the v1.97.8 tag is pushed only after merge, so no public release exists until then. No data migrations; refusals leave every file exactly as it was.

## Docs Impact
- Files updated: CHANGELOG.md (public 1.97.8 notes), docs/UPDATE-RESCUE.md (bridge marker, via release script), the new design spec under docs/superpowers/specs/.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01SoPX4HCJeAPTfss8shwVQs

---
_Generated by [Claude Code](https://claude.ai/code/session_01SoPX4HCJeAPTfss8shwVQs)_

---

> [!NOTE]
> **Medium Risk**
> Changes core update and CLAUDE.md write paths where mistaken refusal blocks upgrades but avoids silent data loss; extensive tests mitigate false positives from the new edit-detection logic.
> 
> **Overview**
> **v1.97.8** stops updates and background refresh from overwriting lines typed directly into `CLAUDE.md`. Composition now compares the live file to the installed release baseline (activation tag, with `refs/dex/installed` fallback), treats unexplained lines as user-owned, and **refuses** with named lines and a rescue path into `CLAUDE-custom.md`; updates **keep** the file byte-for-byte in `kept`/`kept_reasons`. `recompose_if_needed` (including `force`) uses the same rules. Shipped template text, prior composed snapshots, and custom-block edits are distinguished so normal customization and composer upgrades do not wedge vaults.
> 
> **Doctor** adds quick check `config.claude_direct_edits` for early warning; the composition probe no longer advises a refresh that would refuse when direct edits are present.
> 
> **User Profile** composition fills **Pillars** from `System/pillars.yaml` when the profile omits them (profile still wins; bad YAML fails closed).
> 
> The **composition refresh hook** fixes Linux `stat` ordering, bootstraps `System/.dex/claude-composed-baseline.md` on quiet ticks with bounded retries, and release metadata/docs bump to **1.97.8**.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 34973affa1d4e848bed21c4539061e5eba1bdf34. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: core/update/apply_update.py [hunk 1]
```diff
@@ -235,21 +273,99 @@ def _apply_user_profile(template: bytes, vault_root: Path) -> bytes:
...
+    except (RecomposeUnavailable, CompositionError) as error:
+        # Fail closed without over-firing: with no provable baseline, a live
+        # line only counts as lost when the new output does not carry it.
+        baseline = b""
+        baseline_note = (
+            " (what the installed release's CLAUDE.md should contain could "
+            f"not be proved: {error})"
+        )
+    # A line is lost when neither the baseline nor the new output explains it.
+    lost = user_authored_lines(live, baseline + b"\n" + composed)
```
[the remark this answers is not in the document]
Comment (davekilleen) [reply core/update/apply_update.py]: Confirmed and fixed in f2d1724 — this was a genuine wedge: the baseline is composed with current code, so this PR's own pillars overlay made the previously composed placeholder line read as a user edit on every configured vault, and the post-keep case flagged prior-release wording the same way. A line now counts as a user edit only when nothing shipped explains it: the baseline, the new composed output, the incoming raw template, or any release template in the vault's brain store (read lazily, only when the cheap comparison leaves candidates; an unreadable store excuses nothing, so real edits still fail closed). Regression tests cover both of your scenarios — stale composed wording is overwritten forward, prior-release wording is excused, and a genuine user line is still refused with the file untouched.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: core/utils/claude_composition.py [hunk 2]
```diff
@@ -172,12 +284,27 @@ def recompose_if_needed(vault_root: Path, *, force: bool = False) -> str:
...
+                edited = true_user_edits(live, expected, vault_root)
+                if edited:
+                    count = len(edited)
+                    noun = "line was" if count == 1 else "lines were"
+                    return (
+                        f"unavailable:{count} {noun} edited directly into "
+                        f"{CLAUDE} and would be lost by recomposing; move them "
+                        f"into {CUSTOM} (your protected block) first — Dex can "
+                        "do this for you"
+                    )
```
[the remark this answers is not in the document]
Comment (davekilleen) [reply core/utils/claude_composition.py:385]: Confirmed and fixed in a7a96b3 — this one wedged the everyday flow: reword one line in CLAUDE-custom.md and the refresh refused forever, because lines a previous composition wrote from the older custom block or profile were absent from the new expected composition. The composer now records the exact bytes it last wrote (a runtime snapshot under System/.dex/), and lines it provably produced are never treated as the user's own words; a healthy vault with no snapshot yet records one on its first quiet hook tick, when the live file still byte-matches its expected composition. A drifted vault records nothing, so real direct edits keep full protection — a structural shortcut that excused in-block lines by position was tried and rejected because it couldn't distinguish an old custom line from words typed straight into the block. Regression tests: custom-edit reword refreshes cleanly, bootstrap-on-quiet-tick, no-history fail-closed, and out-of-block direct edits still refused after a custom edit.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: core/utils/claude_composition.py [hunk 3]
```diff
@@ -151,6 +325,7 @@ def recompose_if_needed(vault_root: Path, *, force: bool = False) -> str:
     worse than a stale one.
     """
     if not force and not needs_recompose(vault_root):
+        _bootstrap_snapshot(vault_root)
```
[the remark this answers is not in the document]
Comment (davekilleen) [reply core/utils/claude_composition.py:340]: Confirmed and fixed in 0d52b3b — you were right that the bash gate exits before Python on quiet ticks, so the bootstrap was unreachable in production and my tests proved nothing about the real path. The hook now checks for the missing snapshot on the quiet branch (one extra stat once it exists) and starts Python at most once per vault to record it while the live file still matches its expected composition. New tests execute the actual bash hook end to end: bootstrap-on-quiet-tick, quiet-and-silent once the snapshot exists, and the reword-then-refresh flow through the hook.

Writing those tests also surfaced a pre-existing fault your finding led straight to: GNU stat treats `-f` as file-system mode and succeeds, so the BSD-first mtime read concatenated both outputs and the comparison always errored — the refresh hook has been silently inert on every Linux vault. The GNU form now runs first with BSD as the fallback.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: .claude/hooks/claude-composition-refresh.sh [hunk 4]
```diff
@@ -62,10 +62,28 @@
...
+            # composition — so when that record is missing, start Python once
+            # to write it. One extra stat on every later tick; the Python
+            # start happens at most once per vault.
+            [ -f "$SNAPSHOT_FILE" ] && exit 0
+            (cd "$CLAUDE_DIR" && "${DEX_PYTHON_CMD[@]}" -c '
+from pathlib import Path
+from core.utils.claude_composition import _bootstrap_snapshot
+_bootstrap_snapshot(Path("."))
+' >/dev/null 2>&1) || true
+            exit 0
```
[the remark this answers is not in the document]
Comment (davekilleen) [reply .claude/hooks/claude-composition-refresh.sh:101]: Confirmed and fixed in 449e649 — the once-per-vault bound indeed only held for vaults the bootstrap could satisfy. The bootstrap now leaves an attempt marker whatever happens, and the hook retries only when CLAUDE.md has changed since the last attempt: a drifted vault pays one interpreter start per change to the file (an update healing it earns exactly one fresh attempt, which then records the snapshot), never one per prompt. The regression test drives the real bash hook with a counting interpreter wrapper and pins the start counts through drift, repeated quiet ticks, and healing.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: .claude/hooks/claude-composition-refresh.sh [hunk 5]
```diff
@@ -62,10 +62,35 @@
...
+            # write it. A drifted vault can never produce the record, so the
+            # attempt marker bounds retries: Python starts again only when
+            # CLAUDE.md has changed since the last attempt (an update healing
+            # the file gets exactly one fresh attempt), never once per prompt.
+            [ -f "$SNAPSHOT_FILE" ] && exit 0
+            ATTEMPT_FILE="$SNAPSHOT_FILE.attempted"
+            if [ -f "$ATTEMPT_FILE" ]; then
+                ATTEMPT_MTIME=$(stat -c %Y "$ATTEMPT_FILE" 2>/dev/null || stat -f %m "$ATTEMPT_FILE" 2>/dev/null) || exit 0
+                [ "$CLAUDE_MTIME" -gt "$ATTEMPT_MTIME" ] 2>/dev/null || exit 0
+            fi
```
[the remark this answers is not in the document]
Comment (davekilleen) [reply .claude/hooks/claude-composition-refresh.sh:95]: Confirmed and fixed in 85b67b1 — the marker now earns a retry on a slow clock as well as on file change: the hook attempts a fresh bootstrap when CLAUDE.md changed since the last attempt or when the marker is older than six hours. A transient failure (unreadable brain store, mid-update race, a repair that never touches CLAUDE.md, a same-second heal) self-corrects within the window, while a genuinely drifted vault pays at most four interpreter starts a day. The test drives the real hook with a seven-hour-old marker and pins one retry, the successful snapshot record, and the refreshed marker holding afterwards.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: core/tests/test_claude_composition.py [hunk 6]
```diff
@@ -339,3 +347,631 @@ def test_probe_reports_unknown_when_it_cannot_check(tmp_path):
...
+    )
+
+    # And a FRESH marker still gates: no retry within the window.
+    (root / SNAPSHOT_RELATIVE).unlink()
+    done = subprocess.run(["bash", str(hook)], capture_output=True, text=True,
+                          timeout=60, env=env)
+    assert done.returncode == 0
+    assert len(counter.read_text().splitlines()) == 2, (
+        "removing the snapshot with a stale marker retries once more"
+    )
```
[the remark this answers is not in the document]
Comment (davekilleen) [reply core/tests/test_claude_composition.py:975]: Confirmed and fixed in 34973af. The fixture aged the custom file *newer* than CLAUDE.md, which routed the hook down the recompose path — so the clock condition on the quiet path was never executed, and deleting that clause left the test green.

The test now makes the tick genuinely quiet: the custom file is older than CLAUDE.md, and the attempt marker and CLAUDE.md share a timestamp so the changed-file retry condition can never fire. It pins three things in sequence: the stale marker admits exactly one clock-only retry, the fresh marker then gates further starts, and re-aging the marker past six hours admits one more. Verified by mutation — removing the slow-clock clause from the hook now fails this test.

---
_Generated by [Claude Code](https://claude.ai/code)_
